tests/flytekit/unit/core/test_serialization.py

```diff
@@ -943,3 +943,102 @@ def wf_with_input() -> typing.Optional[typing.List[int]]:
     )
 
     assert wf_with_input() == input_val
+
+def test_positional_args_task():
+    arg1 = 5
+    arg2 = 6
+    ret = 11
+
+    @task
+    def t1(x: int, y: int) -> int:
+        return x + y
+
+    @workflow
+    def wf_pure_positional_args() -> int:
+        return t1(arg1, arg2)
```

Comment:
could you also write a test (and run on serverless) where a downstream task takes both positional and named arguments from both 1) workflow inputs and 2) outputs of upstream tasks?

Reply:
Done. Please see if these tests are what you want. Thanks. Also, I'll run them on the serverless later.

https://github.com/flyteorg/flytekit/pull/2522/files#diff-fa72fab667c8393cca8d5afd3943db0edc55794423b7cfc3ab33d0f8c3146650R1022-R1044